src/PowerShellEditorServices/Session/Host/TerminalPSHostRawUserInterface.cs

```diff
@@ -85,97 +79,67 @@ public override Coordinates CursorPosition
         {
             get
             {
-                return
-                    new Coordinates(
-                        System.Console.CursorLeft,
-                        System.Console.CursorTop);
-            }
-            set
-            {
-                System.Console.CursorLeft = value.X;
-                System.Console.CursorTop = value.Y;
+                return new Coordinates(
+                    ConsoleProxy.GetCursorLeft(),
+                    ConsoleProxy.GetCursorTop());
             }
+
+            set => this.internalRawUI.CursorPosition = value;
         }
 
         /// <summary>
         /// Gets or sets the size of the cursor in the console buffer.
         /// </summary>
         public override int CursorSize
         {
-            get;
-            set;
+            get => this.internalRawUI.CursorSize;
+            set => this.internalRawUI.CursorSize = value;
         }
 
         /// <summary>
         /// Gets or sets the position of the console's window.
         /// </summary>
         public override Coordinates WindowPosition
         {
-            get
-            {
-                return
-                    new Coordinates(
-                        System.Console.WindowLeft,
-                        System.Console.WindowTop);
-            }
-            set
-            {
-                System.Console.WindowLeft = value.X;
-                System.Console.WindowTop = value.Y;
-            }
+            get => this.internalRawUI.WindowPosition;
+            set => this.internalRawUI.WindowPosition = value;
         }
 
         /// <summary>
         /// Gets or sets the size of the console's window.
         /// </summary>
         public override Size WindowSize
         {
-            get
-            {
-                return
-                    new Size(
-                        System.Console.WindowWidth,
-                        System.Console.WindowHeight);
-            }
-            set
-            {
-                System.Console.WindowWidth = value.Width;
-                System.Console.WindowHeight = value.Height;
-            }
+            get => this.internalRawUI.WindowSize;
+            set => this.internalRawUI.WindowSize = value;
         }
 
         /// <summary>
         /// Gets or sets the console window's title.
         /// </summary>
         public override string WindowTitle
         {
-            get;
-            set;
+            get => this.internalRawUI.WindowTitle;
+            set => this.internalRawUI.WindowTitle = value;
         }
 
         /// <summary>
         /// Gets a boolean that determines whether a keypress is available.
         /// </summary>
         public override bool KeyAvailable
         {
-            get { return System.Console.KeyAvailable; }
+            get => this.internalRawUI.KeyAvailable;
```

Comment:
can this just be:

```csharp
public override bool KeyAvailable => this.internalRawUI.KeyAvailable;
```

Reply:
Yeah but it should go through `ConsoleProxy`, forgot about this one 🙂